prow/tekton.md

```diff
@@ -0,0 +1,147 @@
+3. Then when the `ProwJob` is triggered:
+  a. Prow (via [the Prow pipeline controller](#pipeline-and-crier-controllers))
+     will create [a PipelineRun](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/pipelineruns.md)
+  b. When the `PipelineRun` completes, the Prow Pipeline controller will update
+     the `ProwJob`
+  c. Finally, [crier](#pipeline-and-crier-controllers) will see the update to
+     the `ProwJob` and update `GitHub` with the results
+
+### Create and apply `Pipelines` and `Tasks`
+
+Before anything can be executed, Tekton expects
+[`Pipelines`](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/pipelines.md)
+and the [`Tasks`](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/tasks.md)
+that the `Pipelines` reference to exist.
+
+Since [Prow is only compatible with versions 0.2.0 - 0.3.1 of Tekton Pipelines](https://github.com/kubernetes/test-infra/issues/13948)
+the below docs are pinned to v0.3.1 of Tekton Pipelines:
+
+* See [the Tekton tutorial](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/tutorial.md)
+  for an overview
+* See [the Tekton Pipelines docs](https://github.com/tektoncd/pipeline/tree/v0.3.1/docs#tekton-pipelines)
+  for reference docs
+* See [the Tekton catalog](https://github.com/tektoncd/catalog)
+  for examples of `Tasks` you might want to use.
+
+### Add `tekton-pipeline` ProwJobs
+
+To configure a `ProwJob` to use Tekton Pipelines, include a `pipelineRunSpec`,
+for example:
+
+```yaml
+presubmits:
+  bobcatfish/catservice:
+  - name: do-the-pipeline
+    agent: tekton-pipeline # Use the Prow Tekton controller!
+    rerun_command: "/run do-the-pipeline"
+    trigger: "(?m)^/run do-the-pipeline,?(\\s+|$)"
+    always_run: true
+    skip_report: false
+    pipeline_run_spec: # Template for creation of the `PipelineRun`
+      trigger: # Required by versions 0.2.0 - v0.3.1 of Tekton Pipelines
+        type: manual
+      pipelineRef:
+        name: special-pipeline # Use the Tekton Pipeline called `special-pipeline`
+      resources:
+      - name: git
+        resourceRef:
+          name: PROW_IMPLICIT_GIT_REF # Used by Prow to add the triggering git ref
+```
+
+* `agent: tekton-pipeline` - Tells Prow that execution of this job should be
+  handed to [the Prow Pipeline controller](#pipeline-and-crier-controllers)
+* `pipeline_run_spec` - The `spec` field of a
+  [PipelineRun](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/pipelineruns.md),
+  used to instantiate a new `PipelineRun` whenever triggered
+* `trigger` - A
+  [now deprecated field in a `PipelineRun`](https://github.com/tektoncd/pipeline/releases/tag/v0.4.0)
+  which must be included though it does nothing (until
+  [#13948](https://github.com/kubernetes/test-infra/issues/13948)!)
+* `pipelineRef` - Refers to a
+  [`Pipeline`](#create-and-apply-pipelines-and-tasks) that must exist for the
+  `PipelineRun` to be able to execute
+* [`PROW_IMPLICIT_GIT_REF`] - Using this special
```

Comment:
It is probably worth noting that we also support `PROW_EXTRA_GIT_REF_0`, `PROW_EXTRA_GIT_REF_1`, etc. if the ProwJob specifies `extraRefs` e.g. https://github.com/kubernetes/test-infra/blob/6eb5baa91054e17e8c1db7f7cda3537170f6a1be/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L677-L680

Reply:
thanks for mentioning this, this saved my life when I went to use this for periodic jobs XD